samples/CMakeLists.txt

```diff
@@ -9,8 +9,37 @@ set(NOVELRT_SAMPLES_LINK_LIBRARIES
 )
 
 add_executable(NovelRTSamples ${NOVELRT_SAMPLES_HEADERS} ${NOVELRT_SAMPLES_SOURCES})
+add_dependencies(NovelRTSamples NovelRTSamplesDependencies)
+
 target_link_libraries(NovelRTSamples ${NOVELRT_SAMPLES_LINK_LIBRARIES})
 
 add_custom_command(TARGET NovelRTSamples POST_BUILD
-  COMMAND ${CMAKE_COMMAND} -E copy_directory $<TARGET_FILE_DIR:NovelRT> $<TARGET_FILE_DIR:NovelRTSamples>
+  COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/NovelRTSamples.txt
+)
+
+set_property(TARGET NovelRTSamples PROPERTY
+  BINARY_OUTDIR $<TARGET_FILE_DIR:NovelRTSamples>
+)
+
+# NovelRTSamples.Dependencies
+
+set(NOVELRT_SAMPLES_DEPENDENCIES_INPUTS
+  $<TARGET_FILE:NovelRT>
+  ${CMAKE_BINARY_DIR}/src/NovelRTDotNet.txt
+  ${CMAKE_BINARY_DIR}/src/NovelRTResources.txt
 )
+
+set(NOVELRT_SAMPLES_DEPENDENCIES_OUTPUTS
+  ${CMAKE_CURRENT_BINARY_DIR}/NovelRT.txt
+  ${CMAKE_CURRENT_BINARY_DIR}/NovelRTDotNet.txt
+  ${CMAKE_CURRENT_BINARY_DIR}/NovelRTResources.txt
+  ${CMAKE_CURRENT_BINARY_DIR}/NovelRTSamplesDependencies.txt
+)
+
+add_custom_command(OUTPUT ${NOVELRT_SAMPLES_DEPENDENCIES_OUTPUTS}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory $<TARGET_FILE_DIR:NovelRT> $<TARGET_GENEX_EVAL:NovelRTSamples,$<TARGET_PROPERTY:NovelRTSamples,BINARY_OUTDIR>>
```

Comment:
There needed to be a little bit of magic for `NovelRTSamples` to depend on `NovelRTSamplesDependencies` while still allowing `NovelRTSamplesDependencies` to determine where the executable was being placed.